Interface Specification

Kronos3 opened the pull request:
- Interfaces
~~Interface Templates~~ (~~merged into interfaces~~ We decided to not implement this in this round)
- Interface importing into components
~~- Interface Aliases~~

Closes #697.

Kronos3 commented:
After discussion, I have merged interface templates into interfaces. Now interfaces are what we _were_ calling templates without any parameters.

bocchino commented on compiler/lib/src/main/scala/analysis/Analyzers/InterfaceAnalyzer.scala:
```diff
@@ -0,0 +1,20 @@
+package analysis.Analyzers
```
I think this file should be removed from this PR.

bocchino commented on docs/spec/Definitions/Interface-Alias-Definitions.adoc:
```diff
@@ -0,0 +1,38 @@
+=== Interface Alias Definitions
```
I think this file should be removed from this PR.

Kronos3 commented:
Updated the wiki:
- Added interfaces to the analysis homepage
- Check semantics passes updates:
  - Enter symbols
  - Check uses
  - **Added new pass** [Check-Interface-Definitions](https://github.com/nasa/fpp/wiki/Check-Interface-Definitions) (before component definitions pass)
  - Check-Component-Definitions

bocchino reviewed (approved):
Looks good!
